substrate/dex/pallet/src/lib.rs

```diff
@@ -561,11 +537,14 @@ pub mod pallet {
     ) -> DispatchResult {
       let sender = ensure_signed(origin)?;
       ensure!((sri_desired > 0) && (coin_desired > 0), Error::<T>::WrongDesiredAmount);
-      ensure!(coin != Coin::Serai, Error::<T>::EqualCoins);
 
-      let pool_id = Self::get_pool_id(coin, Coin::Serai).unwrap();
+      let pool_id = Self::get_pool_id(coin, Coin::Serai)?;
 
-      Pools::<T>::get(pool_id).as_ref().ok_or(Error::<T>::PoolNotFound)?;
+      // create the pool if it doesn't exist. We can just attempt to do that because our checks
+      // far enough to allow that.
+      if Pools::<T>::get(pool_id).is_none() {
+        Self::create_pool(coin)?;
+      }
```

Comment:
We do call `add_liquidity` before we set genesis is complete. We do call it in process of completing the genesis. So that check would make that call not possible. No one already can't call `add_liquidity` since there is no SRI until genesis is complete.

Reply:
> since there is no SRI until genesis is complete.

That should be a safe invariant to rely on. I was just considering the damage possible if even a single SRI became free-floating somehow.